src/Host/Client/Impl/BrokerServices/RemoteUriWebService.cs

```diff
@@ -47,21 +47,33 @@ public class RemoteUriWebService : IRemoteUriWebService {
                 }
             }
 
-            HttpWebResponse response = (HttpWebResponse)await request.GetResponseAsync();
-
-            if (context.Request.IsWebSocketRequest) {
-                Stream respStream = response.GetResponseStream();
-                string subProtocol = response.Headers[Constants.Headers.SecWebSocketProtocol];
-                var remoteWebSocket = CommonWebSocket.CreateClientWebSocket(respStream, subProtocol, TimeSpan.FromMinutes(10), 65335, true);
-                var websocketContext = await context.AcceptWebSocketAsync(subProtocol, 65335, TimeSpan.FromMinutes(10));
-                await WebSocketHelper.SendReceiveAsync(websocketContext.WebSocket, remoteWebSocket, ct);
-            } else {
-                SetResponseHeaders(response, context.Response, localBaseUrl, remoteBaseUrl);
-                using (Stream respStream = response.GetResponseStream())
-                using (Stream outStream = context.Response.OutputStream) {
-                    await outStream.CopyAndFlushAsync(respStream);
+            HttpWebResponse response = null;
+            try {
+                response = (HttpWebResponse)await request.GetResponseAsync();
+            } catch (WebException wex) {
+                if(wex.Status == WebExceptionStatus.ProtocolError) {
+                    response = wex.Response as HttpWebResponse;
+                } else {
+                    throw wex;
+                }
+            } finally {
+                if (response != null) {
+                    if (context.Request.IsWebSocketRequest) {
```

Comment:
This might do the wrong thing when you have a websocket request and an error response. A successful websocket request _must_ have HTTP response 101 (switching protocols). If it's any other error, the response is not a websocket stream (e.g. because the server to which the request was made doesn't support websockets), but rather just a generic HTTP response. So if you call `AcceptWebSocketAsync` on that, it'll break.